### ISSUE TITLE
untag --all applies to messages in search buffer

### ISSUE THREAD

**Issue by kirelagin** [pazz/alot#1498]
I am not sure about this one, maybe its just me. The semantics of `untag --all` in a search buffer is to apply to all matching _messages_. However I subconsciously expect it to apply to _threads_ that I currently see on the screen. I guess it might be useful to have an option for this.

**Describe the solution you'd like**

Change the behaviour of `untag --all` (and similar commands) to apply to apply to all messages in all threads in which at least one message matches the query.

I.e. `search tag:inbox; untag --all foo` should be equivalent to `notmuch tag -foo 'thread:{tag:inbox}'` rather than `not much tag -foo 'tag:inbox'`. But since the query can be more complex, it might not be possible to put it inside `thread:{}`, so that will likely need to be an external loop.

**Describe alternatives you've considered**

* Do nothing and untag from the terminal or by refining the query to use `thread{}` first.
* Keep `untag --all` as is and add another flag or another command for the new behaviour.

**Additional context**

I think it’s just more natural for the command to apply to objects I currently see on the screen, rather than something implied by what I see. Also it will be more consistent with `untag --all` in a thread buffer, where it applies to all messages I see regardless of how exactly I found this thread originally.

**Comment by pazz** [comment]
I agree with this. In fact the notmuch query syntax `thread:{tag:inbox}` is new to me and it certaily wasn't available when I wrote this code (I remember that everything in notmuch was message based, and lots of people kept complaining about the resulting unintuitive behaviour of alot, which I could not do anything about).
If it is now part of libnotmuch, and propagated to the python bindings, then it should be cheap to implement what you suggest.
A PR would be much appreciated.

**Comment by mjg** [comment]
I agree that this is more natural,but how do you get the current behaviour when you need it? For that we would need a search buffer which displays matching messages only. Then the same `untag --all` in these two different search buffers would give bothe behaviours in a completely natural way.

**Comment by guludo** [comment]
Maybe the requested behavior should be implemented with a different option, like `--all-threads`?
